File: Implementation/python/08_주사위네개.py
n, dice = int(input()), []
result = 0

for i in range(n):
    dice.append(list(map(int, input().split())))
    cnt = len(set(dice[i]))
    dice[i].sort()

    if cnt == 1:
        result = max(result, 50000 + dice[i][0] * 5000)
    elif cnt == 2:
        if dice[i][0] == dice[i][1] and dice[i][2] == dice[i][3]:
            result = max(result, 2000 + dice[i][0] * 500 + dice[i][2] * 500)
        else:
            result = max(result, 10000 + dice[i][1] * 1000)
    elif cnt == 3:
        if dice[i][1] == dice[i][2]:
            result = max(result, 1000 + dice[i][1] * 100)
        elif dice[i][0] == dice[i][1]:
             result = max(result, 1000 + dice[i][1] * 100)
        else:
            result = max(result, 1000 + dice[i][2] * 100)
    else:
        result = max(result, max(dice[i]) * 100)

print(result)

File: Implementation/c++/08_주사위네개.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/2484
// 1

#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main()
{
	int N, result = 0;
	cin >> N;

	for (int i = 0; i < N; i++) {
		vector<int> dice(4);
		for (int j = 0; j < 4; j++)
			cin >> dice[j];
		sort(dice.begin(), dice.end());

		int cnt = 0;
		if (dice[0] == dice[3])
			cnt = 50000 + dice[0] * 5000;
		else if ((dice[0] == dice[1] && dice[1] == dice[2]) || (dice[1] == dice[2] && dice[2] == dice[3]))
			cnt = 10000 + dice[1] * 1000;
		else if (dice[0] == dice[1] && dice[2] == dice[3])
			cnt = 2000 + dice[1] * 500 + dice[2] * 500;
		else if ((dice[0] == dice[1]) || (dice[1] == dice[2]))
			cnt = 1000 + dice[1] * 100;
		else if (dice[2] == dice[3])
			cnt = 1000 + dice[2] * 100;
		else
			cnt = dice[3] * 100;

		result = max(result, cnt);
	}

	cout << result;
}